home/wagtail_hooks.py

```diff
@@ -38,3 +42,16 @@ def sort_footer_page_listing_by_path(parent_page, pages, request):
         pages = pages.order_by('path')
 
     return pages
+
+
+@hooks.register('construct_page_chooser_queryset')
+def limit_page_chooser(pages, request):
+    # TODO: add section check otherwise page chooser will break in other places like banner etc.
```

Comment:
this was implemented just for demonstration purposes as mentioned by Danny in a previous comment and TODO was added to discuss the approach. I think it would be better if you hope on a quick call for this as there a quite a few edge cases that I'm concerned about.